filament/backend/src/metal/MetalState.mm

```diff
@@ -79,9 +79,8 @@
 
     // Depth attachment
     descriptor.depthAttachmentPixelFormat = state.depthStencilAttachmentPixelFormat;
-    const bool hasStencil = (MTLPixelFormatDepth32Float_Stencil8 == state.depthStencilAttachmentPixelFormat)
-                        || (MTLPixelFormatDepth24Unorm_Stencil8 == state.depthStencilAttachmentPixelFormat);
-    if (hasStencil) {
+    const bool hasStencil = isStencilFormat(state.depthStencilAttachmentPixelFormat);
+    if (hasStencil && state.stencilWrite) {
```

Comment:
Aren't `hasStencil` and `state.stencilWrite` both derived from calling `isStencilFormat()` on the same format value? How does this fix the issue?